components/script/dom/bindings/like.rs

```diff
@@ -0,0 +1,301 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#![allow(unsafe_code)]
+
+//! Implementation of `setlike<...>` and `maplike<..., ...>` WebIDL declarations.
+
+use crate::dom::bindings::cell::DomRefCell;
+use indexmap::{IndexMap, IndexSet};
+use js::conversions::ToJSValConvertible;
+use std::cmp::Eq;
+use std::hash::Hash;
+
+use super::iterable::Iterable;
+
+/// Every Setlike dom_struct must implement this to provide access to underlying storage
+/// so codegen can automatically generate all setlike methods
+///
+/// In case you use type that implements Setlike as underlying storage it's recommended to use `setlike` macro.
+// This is somehow analogues to gecko's SetlikeHelpers
```

Comment:
`// This is similar to gecko's generated SetlikeHelpers namespace.`

Reply:
Actually this is nothing like SetlikeHelpers anymore, this needs to be removed.